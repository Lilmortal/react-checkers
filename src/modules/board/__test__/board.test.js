import React from 'react'
import renderer from 'react-test-renderer'
import { fromJS, OrderedMap } from 'immutable'
import Board from '../components/board'

describe('<Board /> ', () => {
  it('should display the first player turn', () => {
    let tiles = OrderedMap()
    const tile = fromJS({allowDraught: false, hasDraught: false, player: undefined, isSelected: false,
      isHighlighted: false, isEnemy: false, isQueen: false, isAbleToEat: false, topLeftTile: undefined,
      topRightTile: undefined, bottomLeftTile: undefined, bottomRightTile: undefined, draught: undefined})

    tiles = tiles.set(0, tile)

    const rendered = renderer.create(<Board playerTurn={1} tiles={tiles} />)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
