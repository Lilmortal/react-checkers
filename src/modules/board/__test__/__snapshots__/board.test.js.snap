exports[`<Board />  should display the first player turn 1`] = `
<div
  className="board">
  <h1>
    Player 
    1
     turn
  </h1>
</div>
`;
